### PR TITLE
chore!: prepare process proposal state

### PR DESCRIPTION
**Pull request by evan-forbes**
## Description

https://github.com/cosmos/cosmos-sdk/pull/15683 and https://github.com/cosmos/cosmos-sdk/pull/15717 are included in v0.46.13, which force us to stop relying on a query context for a specific height since errors are now thrown if there is no matching height in the store. 

This PR is adds a simplified, but identical, approach to loading a branch of the state for prepare and process proposal as v0.47.x. This approach doesn't rely on height, instead it just creates a branch from the lastest commit multistore. 

NOTE: in our version of `PrepareProposal` we don't pass height or chain-id, both of which are requried for basic signature verification. For this approach to work, we need to be sure to not pass a completely empty header when creating the proposal context. It must at least contain a height (anything greater than zero will work tbh) and chain-id. This PR also add a mechanism that provides access to the chain-id so we can do just that.

Closes: #325
blocked by #324

### CODE REVIEW

**Review by cmwaters — APPROVED** [review]
LGTM.

How does the SDK handle the Chain ID part?

File: baseapp/abci.go [hunk 1]
```diff
@@ -53,6 +53,7 @@ func (app *BaseApp) InitChain(req abci.RequestInitChain) (res abci.ResponseInitC
 	// initialize the deliver state and check state with a correct header
 	app.setDeliverState(initHeader)
 	app.setCheckState(initHeader)
+	app.chainID = req.ChainId
```
Comment (evan-forbes) [inline baseapp/abci.go:56]: cc @cmwaters 

> How does the SDK handle the Chain ID part?

Comment (cmwaters) [reply baseapp/abci.go:56]: I meant usually? Does baseapp have access to the chainID in mainline SDK

Comment (evan-forbes) [reply baseapp/abci.go:56]: ahh I see, they pass it in a weird way using options and reloading the genesis during initiallization of the server

https://github.com/cosmos/cosmos-sdk/blob/666c345ad23ddda9523cc5cd1b71187d91c26f34/server/util.go#L460-L469

Comment (evan-forbes) [reply baseapp/abci.go:56]: figured just setting it during init chain is very similar and simpler

**Review by rootulp — APPROVED** [review]

File: baseapp/abci.go (hunk 1 above)
Comment (cmwaters) [reply baseapp/abci.go:56]: Ok

**Pull request merged**